src/edu/psu/sweng500/team8/gui/PencilMarkGridPanel.java

```diff
@@ -0,0 +1,163 @@
+					} catch (BadLocationException e) {
+						e.printStackTrace();
+					}
+
+					textField.setEditable(false);
+
+					textField.setPreferredSize(new Dimension(18 * 3, 18 * 3));
+
+					pencilMarkCell[i][j] = new JPanel();
+
+					pencilMarkCell[i][j].setPreferredSize(new Dimension(18 * 3,
+							18 * 3));
+					pencilMarkCell[i][j].setBorder(null);
+
+					pencilMarkCell[i][j].add(textField);
+					this.add(pencilMarkCell[i][j]);
+
+				} else {
+
+					pencilMarkCell[i][j] = new JPanel();
+
+					pencilMarkCell[i][j].setPreferredSize(new Dimension(18 * 3,
+							18 * 3));
+					pencilMarkCell[i][j].setBorder(null);
+
+					CombinationCell combinationCell = new CombinationCell(cell, this.gameSession);
+					pencilMarkCell[i][j].add(combinationCell);
+					
+					for (int k = 1; k <= 9; k++) {
+						
+						if (this.gameSession.getGameBoard().getCellConstraints(cell).getUsedNumbers().contains(k)) {
+							((JToggleButton)((CombinationCell)pencilMarkCell[i][j].getComponent(0)).getComponent(k-1)).setEnabled(false);	
+						}
+					}
+					
+					this.add(pencilMarkCell[i][j]);
+				}
+			}
+		}
+	}
+
+	public void refresh() {
+
+		for (int i = 0; i < 9; i++) {
+			for (int j = 0; j < 9; j++) {
+
+				Cell cell = this.gameSession.getGameBoard().getCellGrid()
+						.getCell(i, j);
+
+				if (cell.hasNumber()) {
+					/* cell has a number */
+					if (!ValueType.Given.equals(cell.getType())) {
+						/* This cell has a user-entered number */
+						pencilMarkCell[i][j].removeAll();
+
+						JTextField textField = new JTextField();
+
+						textField
+								.setHorizontalAlignment(javax.swing.JTextField.CENTER);
+						textField.setBorder(javax.swing.BorderFactory
+								.createLineBorder(new java.awt.Color(0, 0, 0)));
+						textField.setText(Integer.toString(cell.getNumber()));
+
+						textField.setEditable(false);
+
+						textField
+								.setPreferredSize(new Dimension(18 * 3, 18 * 3));
+
+						pencilMarkCell[i][j].add(textField);
+					} else {
+						/* Do not touch Given numbers */
+					}
+				} else {
+					/* cell does not have a number */
+					if (pencilMarkCell[i][j].getComponent(0) instanceof JTextField) {
+						/* if JTextField is already there, it means the cell got a number and it was deleted afterwards. 
+						 * We need to put the PencilMark back */
+						pencilMarkCell[i][j].removeAll();
+						
+						CombinationCell combinationCell = new CombinationCell(cell, this.gameSession);
+						pencilMarkCell[i][j].add(combinationCell);
+					} else {
+						/* pencilMark buttons should be pre-populated 
+						 * pencilMark buttons should be disabled for given numbers (possibly user-defined numbers) 
+						 * TODO - Talk to the team if we want to disable user-defined numbers for pencil Marks 
+						 * TODO - We may not have to select/unselect every time we refresh, if we keep track of updated cells */
+						
+						for (int k = 1; k <= 9; k++) {
+							
+							if (cell.getPencilMarks().contains(k)) {
```

Comment:
yea, that I won't do. as you might have noticed I put curly brackets to even a single line of code when it comes to conditions. for line of work I am in, making things more human read-able to any level of dev is important. and that's just easier to catch as you skim through the code. and the functionality stays the same anyway.